playground/src/pages/CodingEnvironment.tsx

```diff
@@ -1,109 +1,134 @@
-import React from "react";
+import React, { useState } from "react";
 import { oneDark } from "@codemirror/theme-one-dark";
 import { eclipse } from "@uiw/codemirror-theme-eclipse";
 import ActionButton from "../components/ActionButton";
 import DropDown from "../components/DropDown";
 import CodeMirror from "@uiw/react-codemirror";
-import init, { Outputs, run_program, prove_program } from "miden-wasm";
+import init, {
+  Outputs,
+  run_program,
+  prove_program,
+  verify_program,
+} from "miden-wasm";
 import toast, { Toaster } from "react-hot-toast";
+import { getExample, checkFields } from "../utils/helper_functions";
 
-export async function getExample(example: string) {
-  const inputs = fetch(
-    `https://raw.githubusercontent.com/0xPolygonMiden/examples/main/examples/${example}.inputs`
-  );
-  const masm = fetch(
-    `https://raw.githubusercontent.com/0xPolygonMiden/examples/main/examples/${example}.masm`
-  );
-  return [(await inputs).text(), (await masm).text()];
-}
-
-function CodingEnvironment(): JSX.Element {  
+function CodingEnvironment(): JSX.Element {
   /**
-   * Helper function to check if input contains only numbers.
+   * We check the inputs and return true or false. We allow:
+   * - an empty input, and
+   * - a valid JSON object containing stack_init or advice_tape (or both)
+   * if the values are numbers
    */
-  function checkInputField(jsonField: JSON, key: string) {
-    const jsonInput = jsonField[key as keyof typeof jsonField];
+  function checkInputs(jsonString: string) {
+    if (jsonString === "") {
+      return true;
+    }
```

Comment:
This function and `checkOutputs` are both quite long and I think it would be better to put them into the utils. This could be done if just used toast messages instead of putting the error message in the output. Personally, I think this might be fine. These are checks that run when the user makes a request and they aren't output errors like invalid proofs, execution failure, etc. Alternatively, you could pass `setOutput` into the function so that these could still be moved to utils without changing the current UX

So to summarize - I would prefer to move both `checkInputs` and `checkOutputs` to utils

Reply:
Jep, I think you are right. Let's do that in the next refactoring PR if you don't mind. This PR is already quite big. 

What do you think?

Reply:
Ok